tests/btb/test_hyper_parameter.py

```diff
@@ -166,3 +167,25 @@ def test_deepcopy(self):
 
             # deep copy should have new attributes
             self.assertIsNot(hyp.range, hyp_copy.range)
+
+    def test_init_with_string_param_type(self):
```

Comment:
I would split this as two test cases, valid values and invalid values.
Also, I think that the loop is not needed. One case for each scenario is enough.